runtime/src/iree/hal/device.h

```diff
@@ -154,6 +154,33 @@ enum iree_hal_semaphore_compatibility_bits_t {
       IREE_HAL_SEMAPHORE_COMPATIBILITY_DEVICE_SIGNAL,
 };
 
+// Bitfield specifying flags controlling an async allocation operation.
+typedef uint64_t iree_hal_alloca_flags_t;
+enum iree_hal_alloca_flag_bits_t {
+  IREE_HAL_ALLOCA_FLAG_NONE = 0,
+
+  // Buffer lifetime is indeterminate indicating that the compiler or
```

Comment:
Curious why we need this flag during allocation?

Reply:
This is for the compiler analysis failure or external allocation that is managed with independent lifetimes (something imported from another process, etc) - if the compiler can't determine when it's safe to deallocate something asynchronously this lets it prevent async deallocations in its own code or by the hosting application (since there may be unbalanced retain/releases and nothing is safe).